File: docs/code-examples/box3d_batch.py
"""Log a batch of oriented bounding boxes."""
import rerun as rr
import rerun.experimental as rr2

rr.init("rerun_example_box3d_batch", spawn=True)

rr2.log(
    "batch",
    rr2.Boxes3D(
        centers=[[2, 0, 0], [-2, 0, 0], [0, 0, 2]],
        half_sizes=[[2.0, 2.0, 1.0], [1.0, 1.0, 0.5], [2.0, 0.5, 1.0]],
        rotations=[
            rr2.cmp.Rotation3D.identity(),
            rr2.dt.Quaternion(xyzw=[0.0, 0.0, 0.382683, 0.923880]),  # 45 degrees around Z
            rr2.dt.RotationAxisAngle(axis=[0, 1, 0], angle=rr2.dt.Angle(deg=30)),
        ],
        radii=0.025,
        colors=[(255, 0, 0), (0, 255, 0), (0, 0, 255)],
        labels=["red", "green", "blue"],
    ),
)

File: docs/snippets/all/archetypes/box3d_batch.cpp
// Log a batch of oriented bounding boxes.

#include <rerun.hpp>

int main() {
    const auto rec = rerun::RecordingStream("rerun_example_box3d_batch");
    rec.spawn().exit_on_failure();

    rec.log(
        "batch",
        rerun::Boxes3D::from_centers_and_half_sizes(
            {{2.0f, 0.0f, 0.0f}, {-2.0f, 0.0f, 0.0f}, {0.0f, 0.0f, 2.0f}},
            {{2.0f, 2.0f, 1.0f}, {1.0f, 1.0f, 0.5f}, {2.0f, 0.5f, 1.0f}}
        )
            .with_rotations({
                rerun::Quaternion::IDENTITY,
                // 45 degrees around Z
                rerun::Quaternion::from_xyzw(0.0f, 0.0f, 0.382683f, 0.923880f),
                rerun::RotationAxisAngle({0.0f, 1.0f, 0.0f}, rerun::Angle::degrees(30.0f)),
            })
            .with_radii({0.025f})
            .with_colors({
                rerun::Rgba32(255, 0, 0),
                rerun::Rgba32(0, 255, 0),
                rerun::Rgba32(0, 0, 255),
            })
            .with_labels({"red", "green", "blue"})
    );
}
